src/esssans/isis/io.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+"""
+File loading functions for ISIS data, NOT using Mantid.
+"""
+from typing import NewType, TypeVar
+
+import sciline
+import scipp as sc
+
+from ..types import RunType
+
+PixelMaskFilename = NewType('PixelMaskFilename', str)
+CalibrationFilename = NewType('CalibrationFilename', str)
+
+FilenameType = TypeVar('FilenameType', bound=str)
+
+
+DataFolder = NewType('DataFolder', str)
+
+
+class Path(sciline.Scope[FilenameType, str], str):
+    """Path to a file"""
+
+
+class Filename(sciline.Scope[RunType, str], str):
+    """Filename of a run"""
+
+
+MaskedDetectorIDs = NewType('MaskedDetectorIDs', sc.Variable)
+"""1-D variable listing all masked detector IDs."""
+
+
+def to_path(filename: FilenameType, path: DataFolder) -> Path[FilenameType]:
+    return f'{path}/{filename}'
+
+
+def read_xml_detector_masking(filename: Path[PixelMaskFilename]) -> MaskedDetectorIDs:
```

Comment:
Is the format of those xml mask files specific to Mantid or are they also used by other software? If the former, should this move into `mantidio`?

Reply:
Well, the XML is used elsewhere, but the files are afaik pecific to Mantid.